internal/cmd/controller/target/target.go

```diff
@@ -59,6 +59,8 @@ func (t *Target) BundleDeployment() *fleet.BundleDeployment {
 	}
 	bd.Spec.Paused = t.IsPaused()
 
+	initialiseOptionsMaps(bd)
```

Comment:
Since this is  a new bundledeployment, I would just initialize the maps, without a function and a nil check. Am I missing something?

Reply:
The issue here is that those maps are set on [L58](https://github.com/rancher/fleet/pull/3052/files#diff-d81320afcb590e95a0b56577f6c5907ea071596efc14d1b09783b552ee5d1de9R58) through `t.Deployment.Spec`, which _may_ already contain values set at the root of `fleet.yaml` (which may then be overridden by `bundleTarget` values).
The panic only occurred. ie the maps were only empty, when no default, all-targets values are specified at the root of `fleet.yaml`. See https://github.com/rancher/fleet-test-data/pull/23 for an example.